polynote-frontend/polynote/ui/util/tags.ts

```diff
@@ -99,6 +99,24 @@ export function span(classes: string[], content: Content) {
     return tag('span', classes, undefined, content);
 }
 
+export function img(classes: string[], src: string, alt: string) {
+    return tag('img', classes, {src, alt}, []);
+}
+
+export function icon(classes: string[], iconName: string, alt?: string) {
+    return span(classes, img(['icon'], `/style/icons/fa/${iconName}.svg`, alt || iconName))
+}
+
+// This is supposed to create an inline SVG so we can style it with CSS. I'm not sure why it doesn't work though...
+// export function icon(classes: string[], iconName: string, alt: string) {
```

Comment:
@jeremyrsmith not sure if you have any ideas here. I was hoping to be able to style the SVGs with CSS by rendering them as external URIs in a `<use>` tag, but I wasn't able to get SVGs to render at all with this approach - seems like they weren't being fetched from the server... 

Didn't want to spend too much time on this as it's pretty easy to just add simple color variants (and I only needed to do that once or twice). But, it's annoying.

Reply:
I'm not familiar with the `use` tag, but it sounds like an XML-specific thing. Maybe it would work if we were using XHTML?

We could use an XMLHttpRequest to fetch the SVG and then just jam it into the document. /shrug